resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -349,13 +349,22 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
       val offerMem = getResource(offer.getResourcesList, "mem")
       val offerCpus = getResource(offer.getResourcesList, "cpus")
       val offerPorts = getRangeResource(offer.getResourcesList, "ports")
+      val offerAllocationInfo = offer.getAllocationInfo
+      val offerReservationInfo = offer
+        .getResourcesList
+        .asScala
+        .find(resource => Option(resource.getReservation).isDefined)
```

Comment:
could you not just use `resource.hasReservation` here instead of defining a whole new method?

Reply:
I had it that why before but it is repeated in two places so I thought to make it a method.